src/coreclr/src/gc/gcee.cpp

```diff
@@ -401,12 +417,18 @@ void GCHeap::DiagTraceGCSegments()
             FIRE_EVENT(GCCreateSegment_V1, address, size, static_cast<uint32_t>(type));
         }
 
-        // large obj segments
-        for (seg = generation_start_segment (h->generation_of (loh_generation)); seg != 0; seg = heap_segment_next(seg))
+        // uoh segments
+        for (int i = uoh_start_generation; i < total_generation_count; i++)
         {
-            uint8_t* address = heap_segment_mem (seg);
-            size_t size = heap_segment_reserved (seg) - heap_segment_mem (seg);
-            FIRE_EVENT(GCCreateSegment_V1, address, size, static_cast<uint32_t>(gc_etw_segment_large_object_heap));
+            for (seg = generation_start_segment (h->generation_of (i)); seg != 0; seg = heap_segment_next(seg))
+            {
+                uint8_t* address = heap_segment_mem (seg);
+                size_t size = heap_segment_reserved (seg) - heap_segment_mem (seg);
+                gc_etw_segment_type segment_type = (i == 3) ?
```

Comment:
> gc_etw_segment_type segment_type = (i == 3) ? [](start = 15, length = 46)

why not do (i == loh_generation) here?

Reply:
Just did not think about it. It does seem odd.